src/main/webapp/populateData.js

```diff
@@ -162,6 +165,22 @@ function populateCalendar() {
 }
 
 /**
+ * Convert the time in hours and minutes,
+ * and concatenate the string to be displayed
+ * @param {int} timeMilli the time in milliseconds that needs to be converted
+ * @return {String} String representing the converted free time
+ */
+function convertTime(timeMilli) {
+  const hours = Math.floor(timeMilli / (60 * 60 * 1000));
+  const minutes = Math.floor((timeMilli - hours * 60 * 60 * 1000)/(60*1000));
```

Comment:
You could clean this up by defining some constants for the unit conversions.

For the minutes calculation, it would be cleaner to use `(timeMilli % HOUR_IN_MILLIS)` instead of subtracting hours.

Reply:
Moved this logic to the server side. I agree that the modulo would be better than the subtraction. We still need to divide by one minute in millisecods, otherwise the value would be minutes, seconds and milliseconds. `Resolved`